tests/test-cluster.py

```diff
@@ -463,3 +465,121 @@ def test_no_cert_reissue_in_nodes(self):
         for vm in self.VM:
             lock_files = vm.run("ls /var/snap/microk8s/current/var/lock/")
             assert "no-cert-reissue" in lock_files.decode()
+
+
+class TestUpgradeCluster(object):
+    @pytest.fixture(autouse=True, scope="module")
+    def setup_old_versioned_cluster(self):
+        """
+        Provision VMs of the previous version and form the cluster.
+        """
+        try:
+            print("Setting up cluster of an older version")
+            if channel_to_test.startswith("latest") or "/" not in channel_to_test:
+                attempt = 0
+                release_url = "https://dl.k8s.io/release/stable.txt"
+                while attempt < 10:
+                    r = requests.get(release_url)
+                    if r.status_code == 200:
+                        last_stable_str = r.content.decode().strip()
+                        last_stable_str = last_stable_str.replace("v", "")
+                        last_stable_str = ".".join(last_stable_str.split(".")[:2])
+                        break
+                    else:
+                        time.sleep(3)
+                        attempt += 1
+
```

Comment:
If we do not manage to get to https://dl.k8s.io/release/stable.txt we should not go on with the test.

Reply:
Makes sense. Fixed.